File: system-test/dotnet/SystemTests/Latest/ExternalSystemContractTests/Clock/ClockStubContractIsolatedTest.cs
using Dsl.Port;
using SystemTests.Latest.ExternalSystemContractTests.Base;

namespace SystemTests.Latest.ExternalSystemContractTests.Clock;

[Collection("Isolated")]
[Trait("Category", "isolated")]
public class ClockStubContractIsolatedTest : BaseExternalSystemContractTest
{
    protected override ExternalSystemMode? FixedExternalSystemMode => ExternalSystemMode.Stub;

    [Fact]
    public async Task ShouldBeAbleToGetConfiguredTime()
    {
        (await Scenario()
            .Given().Clock().WithTime("2024-01-02T09:00:00Z")
            .Then().Clock())
            .HasTime("2024-01-02T09:00:00Z");
    }
}

File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/latest/contract/clock/ClockStubContractIsolatedTest.java
package com.mycompany.myshop.systemtest.latest.contract.clock;

import com.mycompany.myshop.systemtest.configuration.ExternalSystemMode;
import com.mycompany.myshop.systemtest.latest.contract.base.BaseExternalSystemContractTest;
import com.optivem.testing.Isolated;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@Isolated
class ClockStubContractIsolatedTest extends BaseExternalSystemContractTest {
    @Override
    protected ExternalSystemMode getFixedExternalSystemMode() {
        return ExternalSystemMode.STUB;
    }

    @Test
    void shouldBeAbleToGetConfiguredTime() {
        assertDoesNotThrow(() -> scenario
                .given().clock().withTime("2024-01-02T09:00:00Z")
                .then().clock().hasTime("2024-01-02T09:00:00Z"));
    }
}
